configure.ac

```diff
@@ -56,6 +67,14 @@ darwin*)
    my_htop_platform=darwin
    AC_DEFINE([HTOP_DARWIN], [], [Building for Darwin.])
    ;;
+ios6*)
+   # iOS 6 is really a special case of Darwin. We want to reuse existing logic for Darwin as much as
+   # possible, so we are not giving `my_htop_platform` a different value.
+   my_htop_platform=darwin
+   htop_ios6_platform=yes
+   AC_DEFINE([HTOP_DARWIN], [], [Building for Darwin.])
+   AC_DEFINE([HTOP_DARWIN_IOS6], [], [Building for Darwin (iOS 6).])
+   ;;
```

Comment:
I'm not impressed with this. You surely have no idea of how Autoconf-powered `configure` works.
If it's merely about workarounds of certain missing headers or libraries, then you should check those headers/libraries individually and use the relevant macros for those headers/libraries.

Otherwise we could end up unmaintainable code where there are macros of iOS 6, iOS 7, iOS N... everywhere.

Reply:
I see what you mean. You are saying that

1. you'd rather `my_htop_platform` stay generic at the Darwin level and not expose OS-version-specific macros, and that
2. I should figure out exactly what libraries / headers need special treatment for iOS 6 and just have flags for each of them.

Those make sense to me. Admittedly I only write C code at work and we use a very different build system than Autoconfig. Let me read around and see how that can be done.

Reply:
Never mind about this. I removed this branch entirely.